app/src/main/java/com/beemdevelopment/aegis/otp/GoogleAuthInfo.java

```diff
@@ -343,5 +353,30 @@ public int getBatchIndex() {
         public int getBatchId() {
             return _batchId;
         }
+
+        public static List<Integer> getMissingIndexes(@NonNull List<Export> exports) throws IllegalArgumentException {
+            if (Objects.isNull(exports) || exports.isEmpty()) {
+                throw new IllegalArgumentException("Provided list was null or empty.");
+            }
+            int batchId = exports.get(0).getBatchId();
+            int batchSize = exports.get(0).getBatchSize();
+            List<Integer> indexesPresent = new ArrayList<>();
+            List<Integer> indexesMissing = new ArrayList<>();
+
+            for (Export export: exports) {
+                if (export.getBatchId() == batchId) {
+                    indexesPresent.add(export.getBatchIndex());
+                } else {
+                    throw new IllegalArgumentException("Export contains information for an unrelated batch. Try importing 1 batch at a time.");
+                }
+            }
+            for (int i = 0; i < batchSize; i++) {
+                if (!indexesPresent.contains(i)) {
+                    indexesMissing.add(i);
+                }
+            }
+
+            return indexesMissing;
```

Comment:
Something like this might be a bit easier to read:

```java
Set<Integer> indices = exports.stream()
        .map(Export::getBatchIndex)
        .collect(Collectors.toSet());

for (int i = 0; i < exports.get(0).getBatchSize(); i++) {
    if (!indices.contains(i)) {
        missingIndices.add(i);
    }
}

return missingIndices;
```